pkg/cluster/cluster.go

```diff
@@ -211,11 +211,10 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 		if needDeleteCluster {
 			c.logger.Infof("deleting cluster")
 			c.delete()
+			go c.reportFailedStatus()
```

Comment:
The right logic is to set cluster TPR status to fail when cluster fails. It should be straightforward and obvious.